components/onboarding/onboardingContext.js

```diff
@@ -0,0 +1,204 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { driver } from 'driver.js'; // import driver.js
+
+import 'driver.js/dist/driver.css'; //import css
+import onboardingScriptGeneric from './onboardingScriptGeneric.json';
+import onboardingScriptCustom from './onboardingScriptCustom.json';
+
+const OnboardingContext = createContext(undefined);
+
+function useOnboardingProvider() {
+    const router = useRouter();
+    const [active, setActive] = useState(false);
+
+    const [currentRoute, setCurrentRoute] = useState('');
+    const [nextRouteName, setNextRouteName] = useState('');
+
+    const [prevRouteName, setPrevRouteName] = useState('');
+
+    const [currentRouteIndex, setCurrentRouteIndex] = useState(0);
+
+    const [currentSteps, setCurrentSteps] = useState([]);
+
+    const [currentStep, setCurrentStep] = useState(0);
+    const [hasPrev, setHasPrev] = useState(false);
+    const [hasNext, setHasNext] = useState(false);
+    const [currentStepDescription, setCurrentStepDescription] = useState('');
+    const [currentStepTitle, setCurrentStepTitle] = useState('');
```

Comment:
I'm half thinking maybe we should just create one `useImmer` state for this? then we could just have one object like this:
```
const [route, setRoute] = useImmer({
   name: "chat",
   nextName: "account"
   index: 1,
   currentSteps: [...],
   ....
})

then we wouldn't have so many odovidual states, and the consumer of the context could just get `onboarding.route` and have all the informations for the current route